latest/index.html

```diff
@@ -920,6 +940,60 @@ <h3>Data Extension Responses</h3>
+          <li>Reconciliation services SHOULD guard against malicious code when querying their underlying data store with data from <a>reconciliation queries</a> to avoid code [[injection]].</li>
+          <li>Reconciliation services SHOULD NOT parse the JSON requests sent by clients using a code execution mechanism like JavaScript's <code>eval()</code>, since requests could contain malicious code.</li>
+          <li>Reconciliation clients that display data from <a href="#preview-service">preview services</a> SHOULD ensure that the data is encoded, sanitized, or isolated to prevent [[XSS]] and [[CSRF]] attacks.</li>
+          <li>Reconciliation clients SHOULD provide a configuration option to disallow calls to localhost, with the default set to disallow, as a malicious service could cause the client to make arbitrary POST requests to itself.</li>
+        </ul>
+      </section>
+
+      <section>
+        <h3>Other considerations</h3>
+        <ul>
+          <li>Reconciliation services and clients MAY support user authentication (see <a href="#authentication">authentication</a>).</li>
+          <li>Reconciliation features don't expose information about the underlying platform to clients or services, they don't allow clients or services to send data to the underlying platform, access other devices, device sensors, 
+            a user agent's native UI, or enable clients or services to downgrade default security protections.
+          </li>
+        </ul>
+      </section>
+
+    </section>
+
+    <section>
+      <h2>Privacy Considerations</h2>
+
+      <section>
+        <h3>Network data</h3>
+        <ul>
+          <li>All data in <a>reconciliation queries</a> is visible to the reconciliation service, to be processed for returning <a>reconciliation results</a> to the client.
+            To prevent malicious actors from intercepting the data that is sent and received, services SHOULD provide secure HTTPS endpoints for reconciliation.
+          </li>
+          <li>User selections of matching <a>reconciliation candidates</a> provide user-verified statements about the data. These selections are currently not sent back to the reconciliation service
+            (but that may change with <a href="https://github.com/reconciliation-api/specs/issues/30">#30</a>).
+          </li>
+        </ul>
+      </section>
+
+      <section>
+        <h3>Persistent data</h3>
+        <ul>
+          <li>Reconciliation clients MAY introduce new state that persists across browsing sessions for persisting <a>reconciliation queries</a>,
+            the service's <a>reconciliation results</a>, and/or the user's selection of matches.</li>
+          <li>Reconciliation clients running in a browser’s <em>Private Browsing</em> or <em>Incognito</em> mode SHOULD NOT store <a>reconciliation queries</a>, the service's <a>reconciliation results</a>, or the user's selection of matches.
```

Comment:
> [detecting Private browsing/Incognito mode seems to be quite fiddly](https://stackoverflow.com/questions/2860879/detecting-if-a-browser-is-using-private-browsing-mode) [...] Could it be that you got this idea from W3C recommendations which are primarily designed for new DOM/Javascript APIs?

Ah, interesting. Yes I got this from the [W3C questionnaire](https://www.w3.org/TR/security-privacy-questionnaire/), and I would have expected private mode to be a normal thing to check via browser API, but that's actually considered [leaking of private mode](https://bugs.webkit.org/show_bug.cgi?id=157010#c0) (though it still seems [to happen](https://security.stackexchange.com/a/227865)). And I guess it makes sense to consider private mode a purely first-party thing, which is a matter of the browser only, and no business of any site visited.

Somewhere in between I guess are browser plugins, where checking for private mode  [seems to be possible](https://stackoverflow.com/a/22623958). And I guess there it can still provide privacy too: if only the browser part of the plugin handles that information and never sends it to the service. So maybe we could clarify that section by adding something like "For reconciliation clients running as web applications, this is typically handled by the browser, and data will be stored on the web application's server side only."

It's a bit tricky with OpenRefine being a reconciliation client, running as a client-server application, with the client part running in the browser, and the server part often also running on the client machine. But I think with the added wording above we cover considering the issue in general (which for browser plugin clients might be important), and also that in clients like OpenRefine, the client's server part does actually store the data, even if the client part is running in private mode.

Reply:
> For reconciliation clients running as web applications, this is typically handled by the browser, and data will be stored on the web application's server side only.

I am happy with this formulation, but what about reconciliation clients not running as web applications? For desktop software or mobile apps, there is no notion of private browsing, right?

So it seems like:
* for clients which are web applications (and can therefore be run in private browsing), they do not need to worry about this since the private browsing is handled by the browser
* for clients which are not web applications, the notion of private browsing does not even apply in the first place

-> so perhaps we don't need that paragraph at all? Or did I miss something?

Reply:
As we discussed in [today's meeting](https://etherpad.lobid.org/p/reconciliation-october-2022), the remaining thing are browser extensions, which seem to be disabled in private mode by default, but can be enabled both in [Firefox](https://support.mozilla.org/en-US/kb/extensions-private-browsing) and [Chrome](https://www.google.com/url?sa=t&rct=j&q=&esrc=s&source=web&cd=&cad=rja&uact=8&ved=2ahUKEwjx3Zz31936AhUHiv0HHXsEBjQQFnoECBcQAQ&url=https%3A%2F%2Fsupport.google.com%2Fchrome%2Fa%2Fthread%2F140546132%2Fforce-install-extension-in-incognito-mode%3Fhl%3Den&usg=AOvVaw3VVk7Z2pXqqMxKa6CBSC-s). I've made the private mode consideration specific to extensions in https://github.com/reconciliation-api/specs/pull/99/commits/9b27f2179bf1b89a4f3fa7a1750dd5c8806e6629.